src/J2N/Text/StringFormatter.cs

```diff
@@ -95,10 +95,12 @@ protected virtual CultureInfo Culture
                         return CultureInfo.CurrentCulture;
                     case CultureType.CurrentUICulture:
                         return CultureInfo.CurrentUICulture;
+#if !NET40
                     case CultureType.DefaultThreadCurrentCulture:
                         return CultureInfo.DefaultThreadCurrentCulture;
                     case CultureType.DefaultThreadCurrentUICulture:
```

Comment:
I didn’t understand what you meant by the two comments in `StringFormatter.cs`, as I understand it, it’s enough to add the `FEATURE_CULTUREINFO_DEFAULTTHREADCURRENTCULTURES` as `DefineConstants`, with the `'$(TargetFramework)' != 'net40'` condition.

Reply:
You are correct, it would be enough to add a single feature for both *at present*. However, it is possible in some future version of .NET Microsoft will remove either `CultureInfo.CurrentUICulture` or `CultureInfo.DefaultThreadCurrentCulture` and we would need them separate again in that case. IMO, it would be better to make them separate features now, so the only thing that would need to change in the future is the `Directory.Build.targets` file if one of them is removed from the API again.

Reply:
I'm doing 2 `DefineConstants`, one on CultureInfo.CurrentUICulture, the other on CultureInfo.DefaultThreadCurrentCulture and CultureInfo.DefaultThreadCurrentUICulture, right?

Reply:
You are right. Reviewing your changes again, I mistakenly used `FEATURE_CULTUREINFO_CURRENTUICULTURE` in my comment, which we will not need.

Instead the 2 features should be `FEATURE_CULTUREINFO_DEFAULTTHREADCURRENTCULTURE` and  `FEATURE_CULTUREINFO_DEFAULTTHREADCURRENTUICULTURE`.

Reply:
Okay, i will fix this in the next commit.